ServerCore/Pages/Teams/Signup.cshtml.cs

```diff
@@ -0,0 +1,51 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+using System.Threading.Tasks;
+
+namespace ServerCore.Pages.Teams
+{
+    public class SignupModel : EventSpecificPageModel
+    {
+        public bool IsMicrosoft { get; set; }
+
+        public SignupModel(PuzzleServerContext serverContext, UserManager<IdentityUser> manager) : base(serverContext, manager)
+        {
+        }
+
+        public async Task<IActionResult> OnGetAsync()
+        {
+            if (LoggedInUser == null)
+            {
+                return Challenge();
+            }
+
+            if ((EventRole != EventRole.play && EventRole != EventRole.admin)
```

Comment:
Why do admins sign up?

Reply:
Or why NOT authors?

Reply:
Admins create test teams or fix signup problems on site. Authors can't currently do those things